pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,283 @@
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	Status PriorityLevelConfigurationStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevelConfigurationList is a list of PriorityLevelConfiguration objects.
+type PriorityLevelConfigurationList struct {
+	metav1.TypeMeta
+	// `metadata` is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta
+	// `items` is a list of request-priorities.
+	Items []PriorityLevelConfiguration
+}
+
+// PriorityLevelConfigurationSpec is specification of a priority level
+type PriorityLevelConfigurationSpec struct {
+	// `globalDefault` specifies whether this priority level should be considered as the default priority for requests
+	// that do not have any priority level. Only one PriorityClass should be marked as `globalDefault`.
+	// +optional
+	GlobalDefault bool
+	// `exempt` defines whether the priority level is exempted or not.  There should be at most one exempt priority level.
+	// Being exempt means that requests of that priority are not subject to concurrency limits (and thus are never queued)
+	// and do not detract from the concurrency available for non-exempt requests. In a more sophisticated system, the
+	// exempt priority level would be the highest priority level. The field is default to false and only those system
+	// preset priority level can be exempt.
+	// +optional
+	Exempt bool
+	// `assuredConcurrencyShares` is a positive number for a non-exempt priority level, representing the weight by which
+	// the priority level shares the concurrency from the global limit. The concurrency limit of an apiserver is divided
+	// among the non-exempt priority levels in proportion to their assured concurrency shares. Basically this produces
+	// the assured concurrency value (ACV) for each priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	// +optional
+	AssuredConcurrencyShares int32
+	// `queues` is a number of queues that belong to a non-exempt PriorityLevelConfiguration object. The queues exist
+	// independently at each apiserver. The value must be positive for a non-exempt priority level and setting it to 1
+	// disables shufflesharding and makes the distinguisher method irrelevant.
+	// +optional
+	Queues int32
+	// `handSize` is a small positive number for applying shuffle sharding. When a request arrives at an apiserver the
+	// request flow identifier’s string pair is hashed and the hash value is used to shuffle the queue indices and deal
+	// a hand of the size specified here. If empty, the hand size will the be set to 1.
+	// +optional
+	HandSize int32
+	// `queueLengthLimit` is a length limit applied to each queue belongs to the priority.  The value must be positive
+	// for a non-exempt priority level.
```

Comment:
On further thought, I see no reason to forbid a value of zero here.  It can be treated regularly.  Like any other value for the queue length limit, it is a bound on the length of the queue.  Setting it to zero means the admin does not want requests of this priority level to wait in a queue, they may only be executed immediately and only if this priority level has available concurrency.  I see no reason to forbid this setting.